tests/cfour/psi-ghost-grad/input.dat

```diff
@@ -67,17 +39,35 @@ set geom_maxiter 3
 set mp2_type conv
 set scf_type pk
 
-activate(watdim)
 try:
-    optimize('c4-mp2')
-except psi4.ConvergenceError:
+    optimize('mp2', molecule=watdim)
+except psi4.OptimizationConvergenceError as oce:
+    grad_p4_mA = oce.wfn.gradient()
+
+clean()
+opt_clean()
+
+try:
+    optimize('c4-mp2', molecule=c4watdim)
+except psi4.OptimizationConvergenceError as oce:
+    #c4grad = wfn.gradient()
     pass
+
 compare_matrices(grad_p4_mA, psi4.get_gradient(), 5, 'P4 vs C4 gradient w/ghosts')  #TEST
 
 clean()
 opt_clean()
 
 activate(methdim)
+try:
+    optimize('mp2', return_wfn=True)
+except psi4.OptimizationConvergenceError as oce:
+    grad_p4_mB = oce.wfn.gradient()
```

Comment:
I'm a little confused here; it looks like `grad_p4_mB` is only defined if the exception is raised, but is unconditionally checked below.  Should the check below be guarded to see if `grad_p4_mB` is defined?

Reply:
I think it's fine. The point is to get a non-near-zero gradient, so unless optking gets really good and finishes w/i 3 cycles, the exception should always be raised.